### PR TITLE
Spar Polysemy: In-memory interpreters

### PR DESCRIPTION
**Pull request by isovector**
This PR adds several in-memory interpreters for Spar's polysemy effects.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: services/spar/src/Spar/Sem/SAMLUserStore/Mem.hs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+samlUserStoreToMem :: Sem (SAMLUserStore ': r) a -> Sem r a
+samlUserStoreToMem = (evalState @(Map UserRefOrd UserId) mempty .) $
+  reinterpret $ \case
+    Insert ur uid -> modify $ M.insert (UserRefOrd ur) uid
+    Get ur -> gets $ M.lookup $ UserRefOrd ur
+    GetAnyByIssuer is -> gets $ fmap snd . find (eqIssuer is . fst) . M.toList
+    GetSomeByIssuer is -> gets $ coerce . filter (eqIssuer is . fst) . M.toList
+    DeleteByIssuer is -> modify $ M.filterWithKey (\ref _ -> not $ eqIssuer is ref)
+    -- TODO(sandy): Is it OK to ignore uid here?
```
Comment (isovector) [inline services/spar/src/Spar/Sem/SAMLUserStore/Mem.hs:31]: @fisx is it? :)

File: services/spar/src/Spar/Sem/ScimTokenStore/Mem.hs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import Spar.Sem.ScimTokenStore
+
+scimTokenStoreToMem ::
+  Sem (ScimTokenStore ': r) a ->
+  Sem r a
+scimTokenStoreToMem = (evalState @(Map ScimToken ScimTokenInfo) mempty .) $
+  reinterpret $ \case
+    Insert st sti -> modify $ M.insert st sti
+    Lookup st -> gets $ M.lookup st
+    GetByTeam tid -> gets $ filter ((== tid) . stiTeam) . M.elems
```
Comment (fisx) [inline services/spar/src/Spar/Sem/ScimTokenStore/Mem.hs:19]:
```suggestion
    LookupByTeam tid -> gets $ filter ((== tid) . stiTeam) . M.elems
```

File: services/spar/src/Spar/Sem/AReqIDStore/Mem.hs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            boolTTL False True time
+          Nothing -> pure False
+      StoreVerdictFormat ndt areqid vf -> do
+        now <- Now.get
+        modify $ M.insert areqid (addTime ndt now, vf)
+      GetVerdictFormat areqid -> do
+        gets (M.lookup areqid) >>= \case
+          Just (time, vf) -> do
+            boolTTL Nothing (Just vf) time
+          Nothing -> pure Nothing
```
Comment (fisx) [inline services/spar/src/Spar/Sem/AReqIDStore/Mem.hs:37]: this is a different effect.  see [here](https://github.com/wireapp/wire-server/blob/922ec36f798459adc2a7f11fbeb19f2ccbd264fb/services/spar/src/Spar/Data.hs#L185) vs. [here](https://github.com/wireapp/wire-server/blob/922ec36f798459adc2a7f11fbeb19f2ccbd264fb/services/spar/src/Spar/Data.hs#L255)

**Review by fisx — CHANGES_REQUESTED** [review]

File: services/spar/src/Spar/Sem/ScimTokenStore/Mem.hs (hunk 2 above)
Comment (isovector) [reply services/spar/src/Spar/Sem/ScimTokenStore/Mem.hs:19]: Different PR ok?

**Comment by isovector** [comment]
@fisx PTAL

**Review by fisx — APPROVED** [review]

**Pull request merged**